cli/src/main.rs

```diff
@@ -15,41 +15,36 @@ pub fn main() {
         attempt_recover_seed(matches.value_of("password").map(|s| s.to_string()));
         return;
     }
-
-    let command = matches.value_of("COMMAND");
-    let params = matches
-        .values_of("PARAMS")
-        .map(|v| v.collect())
-        .or(Some(vec![]))
-        .unwrap();
-
-    let maybe_server = matches.value_of("server").map(|s| s.to_string());
-
-    let maybe_data_dir = matches.value_of("data-dir").map(|s| s.to_string());
-
     let seed = matches.value_of("seed").map(|s| s.to_string());
     let maybe_birthday = matches.value_of("birthday");
-
     if seed.is_some() && maybe_birthday.is_none() {
         eprintln!("ERROR!");
         eprintln!(
             "Please specify the wallet birthday (eg. '--birthday 600000') to restore from seed."
         );
-        eprintln!("This should be the block height where the wallet was created. If you don't remember the block height, you can pass '--birthday 0' to scan from the start of the blockchain.");
-        return;
+        panic!("This should be the block height where the wallet was created. If you don't remember the block height, you can pass '--birthday 0' to scan from the start of the blockchain.");
```

Comment:
Could we return with a non-success exit code? I'm of the opinion that a panic indicates something has gone wrong internally, we shouldn't be panicking on invalid user input.

Reply:
I believe returning `Err` from `main` accomplishes this.